src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -394,6 +396,29 @@ yield return
                 }
             }
 
+            private StatementSyntax RemoveOutVariablesFromDeclarationExpressions(StatementSyntax statement,
+                HashSet<string> variablesToRemove)
+            {
+                foreach (var declaration in statement
+                    .DescendantNodes()
+                    .Where(n => n.Kind() == SyntaxKind.DeclarationExpression)
+                    .Cast<DeclarationExpressionSyntax>())
+                {
+                    if (declaration.VariableComponent.Kind() == SyntaxKind.TypedVariableComponent)
+                    {
+                        var variableComponent = (TypedVariableComponentSyntax)declaration.VariableComponent;
+                        var name = ((SingleVariableDesignationSyntax)variableComponent.Designation).Identifier.ValueText;
```

Comment:
I think that's the only syntax you can find under a DeclarationExpression at the moment. Added a check for robustness ;-)